
Note- This is also needed for musl-1.2.5 (aside other fixes)- dilfridge

From 52bead95d2957437c691891fcdc49bd6afccdd49 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 12 Apr 2024 18:45:13 +0200
Subject: Create.c: fix uclibc build

Define FALLOC_FL_ZERO_RANGE if needed as FALLOC_FL_ZERO_RANGE is only
defined for aarch64 on uclibc-ng resulting in the following or1k build
failure since commit 577fd10486d8d1472a6b559066f344ac30a3a391:

Create.c: In function 'write_zeroes_fork':
Create.c:155:35: error: 'FALLOC_FL_ZERO_RANGE' undeclared (first use in this function)
  155 |                 if (fallocate(fd, FALLOC_FL_ZERO_RANGE | FALLOC_FL_KEEP_SIZE,
      |                                   ^~~~~~~~~~~~~~~~~~~~

Fixes:
 - http://autobuild.buildroot.org/results/0e04bcdb591ca5642053e1f7e31384f06581e989

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Signed-off-by: Mariusz Tkaczyk <mariusz.tkaczyk@linux.intel.com>
---
 Create.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Create.c b/Create.c
index 4397ff49..d94253b1 100644
--- a/Create.c
+++ b/Create.c
@@ -32,6 +32,10 @@
 #include	<sys/signalfd.h>
 #include	<sys/wait.h>

+#ifndef FALLOC_FL_ZERO_RANGE
+#define FALLOC_FL_ZERO_RANGE 16
+#endif
+
 static int round_size_and_verify(unsigned long long *size, int chunk)
 {
 	if (*size == 0)
--
cgit 1.2.3-korg


From 7759ceda978aba38861d4846d0c1657465b72f04 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 24 Mar 2024 23:13:32 -0700
Subject: [PATCH] include libgen.h for basename API

Musl does no more provide it via string.h therefore builds with newer
compilers e.g. clang-18 fails due to missing prototype for basename
therefore add libgen.h to included headers list

Upstream-Status: Submitted [https://lore.kernel.org/linux-raid/20240325061537.275811-1-raj.khem@gmail.com/T/#u]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Monitor.c        | 1 +
 platform-intel.c | 1 +
 super-intel.c    | 1 +
 3 files changed, 3 insertions(+)

diff --git a/Monitor.c b/Monitor.c
index 824a69f..e3942e1 100644
--- a/Monitor.c
+++ b/Monitor.c
@@ -26,6 +26,7 @@
 #include	"udev.h"
 #include	"md_p.h"
 #include	"md_u.h"
+#include  <libgen.h>
 #include	<sys/wait.h>
 #include	<limits.h>
 #include	<syslog.h>
diff --git a/platform-intel.c b/platform-intel.c
index ac282bc..5d6687d 100644
--- a/platform-intel.c
+++ b/platform-intel.c
@@ -19,6 +19,7 @@
 #include "mdadm.h"
 #include "platform-intel.h"
 #include "probe_roms.h"
+#include <libgen.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/super-intel.c b/super-intel.c
index dbea235..881dbda 100644
--- a/super-intel.c
+++ b/super-intel.c
@@ -23,6 +23,7 @@
 #include "dlink.h"
 #include "sha1.h"
 #include "platform-intel.h"
+#include <libgen.h>
 #include <values.h>
 #include <scsi/sg.h>
 #include <ctype.h>
--
2.44.0


From 8fa7d3cb96e8833743b635fb198675ad6c020b6e Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Tue, 12 Mar 2024 10:51:51 +0100
Subject: [PATCH] util.c: add limits.h include for NAME_MAX definition

Upstream-Status: Submitted [mariusz.tkaczyk@linux.intel.com,linux-raid@vger.kernel.org]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util.c b/util.c
index b145447..a9cb6c4 100644
--- a/util.c
+++ b/util.c
@@ -36,7 +36,7 @@
 #include	<ctype.h>
 #include	<dirent.h>
 #include	<dlfcn.h>
-
+#include	<limits.h>

 /*
  * following taken from linux/blkpg.h because they aren't
